src/EmployeeList.java

```diff
@@ -0,0 +1,49 @@
+import java.util.Collections;
+
+public class EmployeeList extends TableFrame {
+    private EmployeeEntry sortColumn(EmployeeEntry column) {
+        Collections.sort(column);
+        return column;
+    }
+
+    private EmployeeEntry sortColumnReverse(EmployeeEntry column) {
+        Collections.sort(column, Collections.reverseOrder());
+        return column;
+    }
+
+    private void rateColumn(int columnIndex) {
+        EmployeeEntry originalColumn = getColumn(columnIndex);
+        EmployeeEntry sorted = sortColumn(originalColumn);
+        for (int i = 0; i < sorted.size(); i++) {
+            sorted.get(i).rate = i + 1;
+        }
+    }
+
+    private void rateColumnReverse(int columnIndex) {
+        EmployeeEntry originalColumn = getColumn(columnIndex);
+        EmployeeEntry sorted = (EmployeeEntry) sortColumnReverse(originalColumn);
+        for (int i = 0; i < sorted.size(); i++) {
+            sorted.get(i).rate = i + 1;
+        }
+    }
+    public EmployeeList rateDataList() {
+        rateColumnReverse(0);
+        rateColumn(1);
+        rateColumn(2);
+        rateColumn(3);
+        rateColumnReverse(4);
```

Comment:
1, As I mentioned above in case of column will be need to add the code change required. This is not okey;
2. Seems it duplicate configuration...